### ISSUE TITLE
transaction factory generates blocks that can not be imported with import-blocks

### ISSUE THREAD

**Issue by pepyakin** [paritytech/substrate#2977]
When I try to run `--import-blocks` it just seemingly stucks.

```
    Finished release [optimized] target(s) in 0.92s
     Running `target/release/substrate import-blocks blocks.blocks`
2019-06-28 18:18:28 Importing 1241 blocks
2019-06-28 18:18:28 #0 blocks were added to the queue
2019-06-28 18:18:28 #1000 blocks were added to the queue
```

There is no CPU activity coming from substrate while import so which suggests that there might be an issue with locking or dispatching the tasks.

**Comment by tomaka** [comment]
Ah, my bad. We're using `tokio::run` instead of a method that exits immediately once the future passed as parameter has ended.

**Comment by tomaka** [comment]
The fix is quite easy. Replace `tokio::run` with something like:

```rust
tokio::runtime::Runtime::new()
	.expect("Failed to initialize tokio")
	.block_on(fut)
	.expect("Error while importing blocks");
```

However the other part of the issue would be to add a CI test.

**Comment by pepyakin** [comment]
Hey, thanks for the fix! However, unfortunately, it doesn't work. The symptoms are the same. There is no activity and no results.

**Comment by arkpar** [comment]
Works for me in master. 
```
Initializing Genesis block/state (state: 0xea93…a60c, header-hash: 0x4830…833c)
2019-07-02 12:18:33 Loaded block-time = 4 seconds from genesis on first-launch
2019-07-02 12:18:33 Loading GRANDPA authority set from genesis on what appears to be first startup.
2019-07-02 12:18:33 Importing 1148 blocks
2019-07-02 12:18:33 #0 blocks were added to the queue
2019-07-02 12:18:33 #1000 blocks were added to the queue
2019-07-02 12:18:46 #1000 blocks were imported (#148 left)
2019-07-02 12:18:48 Imported 1147 blocks. Best: #1148
```

**Comment by arkpar** [comment]
@pepyakin try importing to an empty DB

**Comment by pepyakin** [comment]
Yes, I did the `purge-chain` before the import.

Just checked with the latest master:

```
./target/release/substrate purge-chain -y --dev  && time ./target/release/substrate factory --dev --mode MasterTo1 --num 500
./target/release/substrate export-blocks --dev blocks.blocks
./target/release/substrate import-blocks --dev blocks.blocks
```

the result is the same.

**Comment by arkpar** [comment]
Apparently the root issue here is that `factory` command produces invalid blocks.
```
2019-07-08 12:50:15.113 tokio-runtime-worker-0 TRACE sync  Starting import of 1 blocks  (1)
2019-07-08 12:50:15.113 tokio-runtime-worker-0 TRACE sync  Header 0x0136…3829 has 0 logs
2019-07-08 12:50:20.835 tokio-runtime-worker-0 TRACE sync  Verifying 1(0x0136…3829) failed: Header 0x0136ef67f325fd563f0c42acdaaaab5ba3864a83b6ad4db792d56fd0c0dc3829 is unsealed
```

**Comment by pepyakin** [comment]
aha! then it is the factory not the import-blocks!

is it expected behavior @cmichi ?

**Comment by Kianenigma** [comment]
After the recent changes to the extrinsic format, I added a reminder for myself to add more tests to factory and subkey code to make sure they generate extrinsic that are correctly constructed and composed. This issue rings a bell now. 

I will try to look into this soonish. Maybe end up seeing what the issue is with the import as well.

**Comment by tomaka** [comment]
The fact that the operation never stops is probably fixed by #3240 .
